tests/unit/list/dlist.c

```diff
@@ -161,13 +162,86 @@ static inline bool verify_tail_head(sys_dlist_t *list,
 
 	return true;
 }
+
+static enum dlist_perf_stats {PEEK_HEAD_TAIL, INSERT_REMOVE} operation;
+clock_t check_dlist_perf(sys_dlist_t *test_dlist, size_t size)
+{
+	struct container_node node_ii[30] = {0};
+	struct container_node test_node;
+
+	sys_dlist_init(test_dlist);
+	zassert_true(sys_dlist_is_empty(test_dlist), NULL);
+
+	/*avoid that array index is overflow*/
+	size = size < 30 ? size : 30;
+
+	for (int i = 0; i < size; i++) {
+		sys_dlist_append(test_dlist, &node_ii[i].node);
+	}
+
+	clock_t start = 0, finish = 0;
+
+	start = clock();
+	for (int i = 0; i < 10000; i++) {
+		switch (operation) {
+		case PEEK_HEAD_TAIL:
+			sys_dlist_peek_head(test_dlist);
+			sys_dlist_peek_tail(test_dlist);
+			break;
+		case INSERT_REMOVE:
+			sys_dlist_insert(&node_ii[size/2].node,
+					&test_node.node);
+			sys_dlist_remove(&test_node.node);
+			break;
+		default:
+			/*Just running without no operations*/
+			break;
+		}
+	}
+	finish = clock();
+
+	return finish - start;
+}
+
+void test_any_operations(enum dlist_perf_stats op)
```

Comment:
rename this, anything that start with test_ is usually a formal test that is called by ztest suite.

Reply:
Hi @nashif , thanks for you comments, and I have renamed this function. 

Also I did some operations else: 
In keeping with upstream/master(C99 type), I have rebased my PR and add some Doxygen tags to describe it.